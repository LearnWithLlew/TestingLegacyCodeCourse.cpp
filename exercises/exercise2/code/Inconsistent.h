#pragma once
#include <cstdlib>
#include <ctime>
#include <string>
#include <iostream>
#include <sstream>
#include <ctime>
#include <chrono>

class Inconsistent
{
public:
    int getIDOriginal()
    {
        srand(time(0));
        return (rand() % 10 + 1);
    }

    int getID2(int input)
    {
	    return (input % 10 + 1);
    }

    int getID(time_t current_time)
    {
	    time_t seed_for_random_number_generation = current_time;
	    srand(seed_for_random_number_generation);
	    int random_number = rand();
    
	    return getID2(random_number);
    }

    int getID()
    {
        time_t current_time = time(0);
        return getID(current_time);
    }

    std::string print(const std::string& name, std::chrono::system_clock::time_point end)
    {
	    std::stringstream stream;
	    std::time_t end_time =
		    std::chrono::system_clock::to_time_t(end);
	    auto time = std::ctime(&end_time);
	    time[strcspn(time, "\n")] = '\0';
	    stream << "started at " << time << " by " << name;
	    return stream.str();
    }

    std::string print(const std::string& name)
    {
	    return print(name, std::chrono::system_clock::now());
    }
};
