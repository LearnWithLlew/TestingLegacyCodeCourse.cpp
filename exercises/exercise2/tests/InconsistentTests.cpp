#include <catch2/catch.hpp>
#include "ApprovalTests.hpp"

#include "Inconsistent.h"

using namespace ApprovalTests;

TEST_CASE("Scrub print")
{
	Inconsistent inc;
	// Tue Nov 24 16:09:57 2020
	Approvals::verify(inc.print("Oscar"), Options(Scrubbers::createRegexScrubber(R"([A-Z][a-z]{2} [A-Z][a-z]{2} \d{2} \d{2}:\d{2}:\d{2} \d{4})", "[time]")));
}

TEST_CASE("Peel print")
{
	Inconsistent inc;
	
	Approvals::verify(inc.print("Oskar", ApprovalTests::DateUtils::createUtcDateTime(2000, 2, 3, 4, 5, 6)));
}

TEST_CASE("Poke GetID")
{
	Inconsistent inc;
	Approvals::verify(inc.getID(0));
}

TEST_CASE("Poke GetID2")
{
	Inconsistent inc;
	Approvals::verify(inc.getID2(5));
}
