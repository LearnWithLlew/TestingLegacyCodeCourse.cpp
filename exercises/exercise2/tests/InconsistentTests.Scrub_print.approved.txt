started at [time] by Oscar
